File: 01_Python/practice/문자열연습2/17_longestword.py
# *******************************
# 가장 긴 단어 찾기
# 
from functools import reduce

data = [
    "   I am a Student",        # ->  Student
    "That elephant is big  ",  # -> elephant
]

def longestWord(sentence):

    return reduce(
        lambda a, b: a if len(a) > len(b) else b, # 단어 1번째와 2번째를 비교해서 반환
        sentence.strip().split(), # 문자열에서 단어를 받아옴, strip()은 필요 없어도 항상 사용하는 습관!
        )

print(list(map(longestWord, data)))

# ↓ 결과 ['Student', 'elephant']

File: 03_Web/03_JavaScript/A_Basic/a35_문자열연습/17_longestword.js
/*******************************
 * 가장 긴 단어 찾기
 */

const input = [
    "I am a Student",     // ->  Student
    "That elephant is big",  // -> elephant

    // 다른 입력 데이터로 테스트 필요하면, 배열에 데이터 더 넣어보고 실행해도 됩니다
]

function longestWord(str){
    return str.trim().split(/\s+/) // 공백 기준 스플릿
    .reduce((prev, cur) => prev.length > cur.length ? prev : cur)
}

input.forEach(e => console.log(longestWord(e)));